skglm/solvers/pn_solver.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
```

Comment:
I would go for a more explicit filename: `prox_newton.py`. The `_solver` can be omitted IMO (we are in the `solvers` folder).